stellar/txbuilder/main.go

```diff
@@ -0,0 +1,49 @@
+package txbuilder
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stellar/go/clients/horizonclient"
+	"github.com/stellar/go/txnbuild"
+	"github.com/stellar/go/xdr"
+)
+
+type Builder struct {
+	BridgeAccount string
+}
+
+func (b *Builder) BuildTransaction(txSource, destination, amount string) (xdr.TransactionEnvelope, error) {
+	client := horizonclient.DefaultTestNetClient
+
+	if txSource == b.BridgeAccount {
+		return xdr.TransactionEnvelope{}, errors.New("bridge account cannot be used as a transaction source")
+	}
+
+	sourceAccount, err := client.AccountDetail(horizonclient.AccountRequest{AccountID: txSource})
+	if err != nil {
+		return xdr.TransactionEnvelope{}, errors.Wrap(err, "error getting account details")
+	}
```

Comment:
💡 It looks like the only reason we need to hit horizon during this request is to get the sequence number of the `txSource`, however we can offload that responsibility to the caller by having them provide the seq number. It would reduce the work that the bridge has to do, and make it harder for someone to use a bridge validator to DOS a horizon.

Reply:
Good idea, I added this for simplicity of this prototype but I was wondering if we could leave it like this. However, providing a seqnum loaded by a client is a better option.

Reply:
I think it would be better for the client to pick the seq number as well. If the user has control over sequence numbers it will reduce the cases where transactions fail due to the sequence number conflicting with another transaction belonging to the user